Add documentation for promoting the NodeLocalCRISocket to Beta (#51503)
kubernetes/website: content/en/docs/tasks/administer-cluster/kubeadm/configure-cgroup-driver.md

Executing the sub commands `init`, `join` and `upgrade` would result in kubeadm
writing the `KubeletConfiguration` as a file under `/var/lib/kubelet/config.yaml`
and passing it to the local node kubelet.

> For each node, kubeadm detects the CRI socket and stores it in the `/var/lib/kubelet/instance-config.yaml` file.

Reviewer: ```suggestion
On each node, kubeadm detects the CRI socket and stores its details into the `/var/lib/kubelet/instance-config.yaml` file.
```

When executing the `init`, `join`, or `upgrade` subcommands, 

> kubeadm patches the `containerRuntimeEndpoint` value from this instance configuration into `/var/lib/kubelet/config.yaml`.

Reviewer 2: sidenote: hopefully we can remove this file entirely some day after this feature is GA
- https://github.com/kubernetes/enhancements/issues/4033
